src/features/defi/providers/osmosis/components/OsmosisManager/Lp/Withdraw/components/Withdraw.tsx

```diff
@@ -276,16 +275,11 @@ export const Withdraw: React.FC<WithdrawProps> = ({
   )
 
   const handleInputChange = (value: string, isFiat?: boolean) => {
-    if (!(lpAsset && lpAssetMarketData)) return
-    const divisor = bnOrZero(lpAssetMarketData.price)
-
-    const _value = isFiat
-      ? bnOrZero(value)
-          .dividedBy(divisor.toFixed() ?? 1)
-          .toString()
-      : value
+    if (!(lpAsset && lpAssetMarketData?.price)) return
+
+    const _value = isFiat ? bnOrZero(value).div(lpAssetMarketData.price).toString() : value
```

Comment:
Wait no, I've actually got an idea where we can use four nested ternaries here 😂